pennylane_lightning/core/src/simulators/lightning_tensor/utils/tncuda_utils/tncuda_helpers.hpp

```diff
@@ -194,4 +194,29 @@ inline auto create_swap_wire_pair_queue(const std::vector<std::size_t> &wires)
     return {local_wires, swap_wires_queue};
 }
 
+/**
+ * @brief Check if the provided MPS has the correct dimension for C++
+ * backend.
+ *
+ * @param MPS_shape_dest Dimension list of destination MPS.
+ * @param MPS_shape_source Dimension list of incoming MPS.
+ */
+inline void
+MPSShapeCheck(const std::vector<std::vector<std::size_t>> &MPS_shape_dest,
+              const std::vector<std::vector<std::size_t>> &MPS_shape_source) {
+
+    if (!(MPS_shape_dest == MPS_shape_source)) {
+        auto MPS_shape_source_str =
+            Pennylane::Util::vector2DToString<std::size_t>(MPS_shape_source);
+        auto MPS_shape_dest_str =
+            Pennylane::Util::vector2DToString<std::size_t>(MPS_shape_dest);
```

Comment:
Why do we really need these vec2str? It seems redundant if users can print these shapes from the user program in case of the "Incorrect incoming MPS sites data shape" error?

Reply:
I thought being friendly with the user because this print would help the user find easily the problem due to the MPS format for cutensornet is too specific.  However, if you request to remove `vector2DToString` I will delete it.  :smile:

Reply:
Let's be consistent with other runtime error messages! I know this is more friendly but the sort of info can be easily pulled up from the user code.

Reply:
Done!